### PR TITLE
USHIFT-1845: improve CI check_contents function in manage_build_cache

### PR DESCRIPTION
**Pull request by eslutsky**

### CODE REVIEW

File: test/bin/manage_build_cache.sh [hunk 1]
```diff
@@ -59,16 +59,20 @@ Options:
 EOF
 }
 
-check_contents() {
+check_contents(){
     local -r src_dir="s3://${AWS_BUCKET_NAME}/${BCH_SUBDIR}/${UNAME_M}/${TAG_SUBDIR}"
-
+    local must_contain_array=("repo.tar" "${VM_POOL_BASENAME}/.*\.iso")
```
Comment (ggiguash) [inline test/bin/manage_build_cache.sh:64]:
```suggestion
    local -r must_contain_array=("repo.tar$" "${VM_POOL_BASENAME}/.*\.iso$")
```

File: test/bin/manage_build_cache.sh [hunk 2]
```diff
@@ -59,16 +59,20 @@ Options:
...
+    
     echo "Checking contents of '${src_dir}'"
-    if "${AWSCLI}" s3 ls "${src_dir}/${VM_POOL_BASENAME}/" | awk '{print $NF}' | grep -Eq '.iso$' ; then
-        if "${AWSCLI}" s3 ls "${src_dir}/" | awk '{print $NF}' | grep -Eq 'repo.tar$' ; then
-            return 0
+    local s3_stdout="$(${AWSCLI} s3 ls $src_dir --recursive)" || return 1
+
+    for item in "${must_contain_array[@]}"; do 
+        if echo $s3_stdout | grep -q -v $item;  then
```
Comment (ggiguash) [inline test/bin/manage_build_cache.sh:70]: I think we must use `-E` option here to enable more precise check for the generic code.

File: test/bin/manage_build_cache.sh [hunk 3]
```diff
@@ -59,16 +59,21 @@ Options:
 EOF
 }
 
-check_contents() {
+check_contents(){
     local -r src_dir="s3://${AWS_BUCKET_NAME}/${BCH_SUBDIR}/${UNAME_M}/${TAG_SUBDIR}"
-
+    local -r must_contain_array=("repo.tar" "${VM_POOL_BASENAME}/.*\.iso")
```
Comment (ggiguash) [inline test/bin/manage_build_cache.sh:64]:
```suggestion
    local -r must_contain_array=("repo.tar$" "${VM_POOL_BASENAME}/.*\.iso$")
```

Comment (pacevedom) [reply test/bin/manage_build_cache.sh:64]: And also escape the `.` in `repo.tar`.

**Comment by ggiguash** [comment]
Test is not mandatory for the changes
/override ci/prow/metal-periodic-test

**Pull request merged**